interfaces/core.go

```diff
@@ -27,22 +27,42 @@ import (
 
 // Plug represents a capacity offered by a snap.
 type Plug struct {
-	Plug      string
-	Snap      string
-	Interface string
-	Attrs     map[string]interface{}
-	Apps      []string
-	Label     string
+	Snap        string                 `json:"snap"`
+	Name        string                 `json:"plug"`
+	Interface   string                 `json:"interface"`
+	Attrs       map[string]interface{} `json:"attrs,omitempty"`
+	Apps        []string               `json:"apps,omitempty"`
+	Label       string                 `json:"label"`
+	Connections []SlotRef              `json:"connections,omitempty"`
+}
+
+// PlugRef is a reference to a plug.
+type PlugRef struct {
+	Snap string `json:"snap"`
+	Name string `json:"plug"`
 }
 
 // Slot represents the potential of a given snap to connect to a plug.
 type Slot struct {
-	Slot      string
-	Snap      string
-	Interface string
-	Attrs     map[string]interface{}
-	Apps      []string
-	Label     string
+	Snap        string                 `json:"snap"`
+	Name        string                 `json:"slot"`
+	Interface   string                 `json:"interface"`
+	Attrs       map[string]interface{} `json:"attrs,omitempty"`
+	Apps        []string               `json:"apps,omitempty"`
+	Label       string                 `json:"label"`
+	Connections []PlugRef              `json:"connections,omitempty"`
+}
+
+// SlotRef is a reference to a slot.
+type SlotRef struct {
+	Snap string `json:"snap"`
+	Name string `json:"slot"`
+}
+
+// InterfaceConnections contains information about all plugs, slots and their connections
```

Comment:
`// Interfaces holds information about a list of plugs and slots, and their connections.`